web-jehlomat/src/Components/Form/PridatUzivateleForm.tsx

```diff
@@ -0,0 +1,157 @@
+import { primaryDark } from '../../utils/colors';
+import { media } from '../../utils/media';
+import { LINKS } from 'utils/links';
+
+interface Props {}
+
+interface Values {
+    email: string;
+}
+
+const validationSchema = yup.object({
+    email: yup.string().email('Vlož validní email').required('Email je povinné pole'),
+});
+
+const ModalFormWrapper = styled(FormWrapper)<{ mobile?: boolean }>`
+    ${props =>
+        props.mobile &&
+        `
+        height: calc(100vh - 100px);
+        justify-content: flex-start;
+    `}
+`;
+
+const ModalFormItem = styled(FormItem)<{ mobile?: boolean }>`
+    font-family: Roboto;
+    margin: 15px 10px;
+    text-align: center;
+    width: 90%;
+
+    @media (min-width: 700px) {
+        width: 85%;
+    }
+    ${props =>
+        props.mobile &&
+        `
+        width: 90%;
+    `}
+
+`;
+
+const ModalFormItemLabel = styled(FormItemLabel)<{ mobile?: boolean }>`
+    text-transform: none;
+    ${props =>
+        props.mobile &&
+        `
+        font-weight: 700;
+        font-size: 18px;
+        text-align: left;
+    `}
+`;
+
+const ModalFormItemDescription = styled(FormItemDescription)<{ mobile?: boolean, below?: boolean }>`
+    color: ${primaryDark};
+    text-align: center;
+    font-size: 18px;
+    @media (min-width: 700px) {
+        width: 85%;
+    }
+    ${props =>
+        props.mobile &&
+        `
+        margin: 3rem 0;
+        font-weight: 500;
+        line-height: 24px;
+        text-align: left;
+        width: 90%;
+
+    `}
+    ${props =>
+        (props.below && props.mobile) &&
+        `
+        font-size: 16px;
+        margin: 0 0 2rem;
+    `}
+`;
+
+
+const PridatUzivateleModal: FC<Props> = () => {
+    const isMobile = useMediaQuery(media.lte('mobile'));
+
+    let history = useHistory();
+    return (
+        <>
+            <Formik
+                initialValues={{ email: '' }}
+                validationSchema={validationSchema}
+                onSubmit={async (values: Values, { setErrors }) => {
+                    try {
+                        const response: AxiosResponse<any> = await authorizedAPI.post('/api/v1/jehlomat/user', values);
+                        const status = response.status;
+
+                        switch (true) {
+                            case /2[0-9][0-9]/g.test(status.toString()): {
+                                history.push(LINKS.userThankYou);
+                                break;
+                            }
+                            case status === 409: {
+                                //for validation error;
+                                setErrors({ email: response.data });
+                                break;
+                            }
+                            default: {
+                                //all others goes to error page; TODO push to error page
+                                break;
+                            }
+                        }
```

Comment:
```suggestion
                        const isSuccess = /2[0-9][0-9]/g.test(status.toString());
                        
                        if (isSuccess) {
                            history.push(LINKS.userThankYou);
                            
                            return;
                        }
                        
                        if (status === 409) {
                            setErrors({ email: response.data });
                            
                            return;
                        }
                        
                        console.log(`Unexpected HTTP status code ${status}`);
                        history.push(LINKS.errorPage); // verify existence of constant
```

Please instead of `switch (tue)` use `if` condition statements, I believe the code would be more readable.